pkg/codegen/nodejs/gen.go

```diff
@@ -1617,6 +1635,133 @@ type namespace struct {
 	children []*namespace
 }
 
+// The Debug method returns a string representation of
+// the namespace for the purposes of debugging.
+func (ns namespace) Debug() string {
+	var children []string
+	for _, child := range ns.children {
+		children = append(children, child.name)
+	}
+	var childrenStr = strings.Join(children, "\t\n")
+	return fmt.Sprintf(
+		"Namespace %s:\n\tTypes: %d\n\tEnums: %d\n\tChildren:\n%s",
+		ns.name,
+		len(ns.types),
+		len(ns.enums),
+		childrenStr,
+	)
+}
+
+// ioContext defines a set of parameters used when generating input/output
+// type definitions. These parameters are stable no matter which directory
+// is getting generated.
+type ioContext struct {
+	mod             *modContext
+	input           bool
+	imports         map[string]codegen.StringSet
+	externalImports codegen.StringSet
+}
+
+// intoIOFiles converts this namespace into one or more files.
+// It recursively builds one file for each node in the tree.
+// If input=true, then it builds input types. Otherwise, it
+// builds output types.
+// The parameters in ctx are stable regardless of the depth of recursion,
+// but parent is expected to change with each recursive call.
+func (ns *namespace) intoIOFiles(ctx *ioContext, parent string) ([]*ioFile, error) {
+	// We generate the input and output namespaces when there are enums, regardless of i
+	// they are empty.
+	if ns == nil {
+		panic("TODO: The caller needs to check if this is nil or not first.")
+	}
+	fmt.Println(ns.Debug())
+	// Declare a new file to store the contents exposed at this directory level.
+	var dirRoot = path.Join(parent, ns.name)
+	var filename string
+	if ctx.input {
+		filename = path.Join(dirRoot, "inputs.ts")
+	} else {
+		filename = path.Join(dirRoot, "outputs.ts")
+	}
+	fmt.Printf("Generating namespace into IO file: %s\n", filename)
+	var file = newIOFile(filename)
+	// We start every file with the header information.
+	ctx.mod.genHeader(file.writer(), ctx.mod.sdkImports(true, false), ctx.externalImports, ctx.imports)
+	// We want to organize the items in the source file by alphabetical order.
+	sort.Slice(ns.types, func(i, j int) bool {
+		return tokenToName(ns.types[i].Token) < tokenToName(ns.types[j].Token)
```

Comment:
This is not a total order, leading to non-determinism. This sort func has been fixed on `master` I think.

Reply:
I rebased off of `master` and pulled in the change, which I noticed is indeed on `master`.